app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -143,65 +143,104 @@ export function testsForProviderType(providerType) {
 
     describe('methods that have a param to specify the block', () => {
       const supportingBlockParam = [
-        { name: 'eth_call', blockParamIndex: 1 },
-        { name: 'eth_getBalance', blockParamIndex: 1 },
-        { name: 'eth_getBlockByNumber', blockParamIndex: 0 },
-        { name: 'eth_getCode', blockParamIndex: 1 },
-        { name: 'eth_getStorageAt', blockParamIndex: 2 },
-        { name: 'eth_getTransactionCount', blockParamIndex: 1 },
+        {
+          name: 'eth_call',
+          blockParamIndex: 1,
+          numberOfParameters: 2,
```

Comment:
I looked up each method in both the [Ethereum JSON-RPC spec](https://ethereum.github.io/execution-apis/api-documentation/), and the [Infura docs](https://docs.infura.io/infura/networks/ethereum/json-rpc-methods). I ensured the configured number of parameters matched the _maximum_ shown in either docs.

They mostly matched already; I think there was a single method that had one less parameter on Infura because it didn't support some option, don't recall which one. And there were some methods exclusive to one or the other. But other than that they matched.